iaso/api/deduplication/filters.py

```diff
@@ -166,23 +163,15 @@ def filter_queryset(self, request, queryset, view):
         start_date = request.query_params.get("start_date")
         end_date = request.query_params.get("end_date")
 
-        date_format = "%d-%m-%Y"
-
         if start_date:
             try:
-                start_date_dt = datetime.strptime(start_date, date_format)
-                start_date_dt = timezone.make_aware(start_date_dt, timezone.get_default_timezone())
-                start_date_dt = start_date_dt.replace(hour=0, minute=0, second=0)
-                queryset = queryset.filter(analyze__created_at__gte=start_date_dt)
+                queryset = queryset.filter(analyze__created_at__gte=date_string_to_start_of_day(start_date))
```

Comment:
We may be able to use a string and only filter on the `date` part of the `datetime` field.

E.g. (assuming the date format is correct):

```python
queryset.filter(analyze__created_at__date__gte="2021-05-23")

queryset.filter(analyze__created_at__date__lte="2021-05-23")
```

This may be easier to read and may save some code.

Reply:
I like the simplicity, but would that take correctly into account the timezone? :thinking:

Reply:
Correct me if I'm wrong but it seems that the incoming format of `request.query_params.get("start_date")` is a date string (i.e. `"%d-%m-%Y"`) without a clearly defined TZ nor any time information.

If that's the case, you'll never be able to get an accurate "TZ aware" date or datetime.

You may as well rely on `__date__`.

Reply:
@kemar You're right, actually ideally the user should have a timezone on their profile, and then we could use that one to generate the "start of day". If we do it correctly, the current code would be easier to refactor I think. Since it's working as is, okay for you to merge like this?

Reply:
@bramj yup OK